java-checks/src/main/java/org/sonar/java/checks/PublicStaticMutableMembersCheck.java

```diff
@@ -220,6 +220,11 @@ private static boolean isAcceptedTypeOrUnmodifiableMethodCall(MethodInvocationTr
   }
 
   private static boolean isAcceptedType(Type type, List<String> accepted) {
+    // In case of broken semantics, the type is unknown and can therefore not be matched against an accepted one.
+    // To avoid raising FPs, we consider that an unknown type is most likely an accepted one.
+    if (type.isUnknown()) {
```

Comment:
This is now redundant with line 219.

Why not dropping the `type.isUnknown() ||` part of method `isAcceptedTypeOrUnmodifiableMethodCall()` ? it anyway calls `isAcceptedType(type, ACCEPTED_TYPES)` right after.